### PR TITLE
Move /core/src/types/state to /core/src/state

### PR DESCRIPTION
**Pull request by theengineear**
@rgbkrk and I discussed moving `state` up a level since `types` is sorta a lie, the folder has more than types, e..g, `refs` has `createHostRef` and such.

The `core/src/records` is also going to do some moving around in a subsequent PR.

### CODE REVIEW

**Comment by theengineear** [comment]
Eh, not so bad :)

File: packages/core/src/state/index.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 // @flow
-import type { CommunicationRecordProps } from "./communication";
-import type { EntitiesRecordProps } from "./entities";
+import type { CommunicationRecordProps } from "./communication/index";
+import type { EntitiesRecordProps } from "./entities/index";
```
Comment (rgbkrk) [inline packages/core/src/state/index.js]: Why do these have to reference `index` directly?

Comment (theengineear) [reply packages/core/src/state/index.js]: hehe, cuz I let my IDE do the refact 😆 changing. thanks!

Comment (rgbkrk) [reply packages/core/src/state/index.js]: Ok, I'll hold off on merging here. 😄

**Comment by theengineear** [comment]
@rgbkrk thanks for the catch!

**Pull request merged**